packages/testing/src/consensus_testing/keys.py

```diff
@@ -0,0 +1,101 @@
+"""XMSS key management utilities for testing."""
+
+from typing import Optional
+
+from lean_spec.subspecs.containers import Attestation, Signature
+from lean_spec.subspecs.containers.slot import Slot
+from lean_spec.subspecs.ssz.hash import hash_tree_root
+from lean_spec.subspecs.xmss.containers import PublicKey, SecretKey
+from lean_spec.subspecs.xmss.interface import DEFAULT_SIGNATURE_SCHEME
+from lean_spec.types import ValidatorIndex
+
+
+class XmssKeyManager:
+    """
+    Manages XMSS keys for test validators.
+
+    Generates and manages XMSS key pairs for validators on demand.
+    Keys are generated to be valid up to the specified max_slot.
+    """
+
+    DEFAULT_MAX_SLOT = Slot(100)
+    """Default maximum slot for key generation."""
+
+    max_slot: Slot
+    public_keys: dict[ValidatorIndex, PublicKey]
+    secret_keys: dict[ValidatorIndex, SecretKey]
+
+    def __init__(
+        self,
+        max_slot: Optional[Slot] = None,
+    ) -> None:
+        """
+        Initialize the XMSS key manager.
+
+        Args:
+            max_slot: Maximum slot for which keys should be valid. Keys will be
+                generated with enough capacity to sign messages up to this slot.
+                Defaults to 100 slots.
+        """
+        self.max_slot = max_slot if max_slot is not None else self.DEFAULT_MAX_SLOT
+        self.public_keys: dict[ValidatorIndex, PublicKey] = {}
+        self.secret_keys: dict[ValidatorIndex, SecretKey] = {}
+
+    def create_key_pair(self, validator_index: ValidatorIndex) -> tuple[PublicKey, SecretKey]:
+        """
+        Create an XMSS key pair for the given validator index.
+
+        Args:
+            validator_index: The index of the validator to create a key for.
+
+        Returns:
+            A tuple containing the public and secret keys for the given validator index.
+        """
+        if validator_index not in self.public_keys:
+            # Use max_slot + 1 as num_active_epochs since slots are used as epochs in the spec.
+            # +1 to include genesis slot
+            num_active_epochs = self.max_slot.as_int() + 1
+            self.public_keys[validator_index], self.secret_keys[validator_index] = (
+                DEFAULT_SIGNATURE_SCHEME.key_gen(0, num_active_epochs)
+            )
```

Comment:
Should the generated privkeys be saved to file somewhere along with the test vectors? I think otherwise clients cannot reproduce and match the test vectors' state and block roots if they're not signing with the same generated keys.